aws/ecs/packer/to-upload/peers.sh

```diff
@@ -1,30 +1,235 @@
 #!/bin/bash
-# This script print a list of IPs of the EC2 instances in your current Autoscaling Group.
+# This script print a list of IPs based on
+#   1) TAG: this will match IPs for instances that have matching values for that Tag key set in /etc/weave/weave.config under
+#        the environment variable WEAVE_AWS_PEER_TAG
+#   2) Autoscaling Group: if this instance is part of an autoscaling group, this will return IPs for its peers
+#   3) Spot Fleet: Finally, if this instance is part of a spot fleet, this will return IPs for all instances in the fleet
 set -eu
 
-aws=/usr/local/bin/aws
+if [ -e '/etc/weave/weave.config' ]; then
+  source /etc/weave/weave.config
+fi
 
-export AWS_DEFAULT_REGION=$(curl -s curl http://169.254.169.254/latest/dynamic/instance-identity/document | \
-                            jq -r .region)
-current_instance_id=$(curl -s curl http://169.254.169.254/latest/meta-data/instance-id)
-current_autoscaling_group=$($aws autoscaling describe-auto-scaling-instances | \
-                            jq -r ".AutoScalingInstances[] | select(.InstanceId == \"$current_instance_id\") | .AutoScalingGroupName")
+WEAVE_AWS_PEER_TAG="${WEAVE_AWS_PEER_TAG:-}"
 
-if [ -z "$current_autoscaling_group" ]; then
-  >&2 echo "instance doesn't belong to an autoscaling group"
-  exit 1
-fi
+__awscli=
+__instance_id=
+__region=
+__instance_autoscaling_group=
+__peer_autoscaling_instance_ids=
+__list_of_active_spot_fleets=
+__spot_fleet_request_id=
+__peer_spot_fleet_instance_ids=
+__peer_private_ips=
+__peer_tag_instance_ids=
 
-peer_instances=$($aws autoscaling describe-auto-scaling-instances | \
-                jq -r ".AutoScalingInstances[] | select(.AutoScalingGroupName == \"$current_autoscaling_group\" and .InstanceId != \"$current_instance_id\") | .InstanceId")
+check_data() {
+  if [[ "${1}x" == "x" ]]; then
+    return 1
+  fi
+}
 
-if [ -z "$peer_instances" ]; then
-# no peers found
-  exit 0
-fi
+check_exit() {
```

Comment:
I don't think we really need this. Where relevant, we should just print what the problem is (why the function failed) instead.